Scripts/Runtime/Data/Collections/UnsafeArray.cs

```diff
@@ -0,0 +1,606 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+using UnityEngine.Internal;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    [NativeContainerSupportsDeferredConvertListToArray]
+    [DebuggerDisplay("Length = {Length}")]
+    [NativeContainerSupportsDeallocateOnJobCompletion]
+    [NativeContainerSupportsMinMaxWriteRestriction]
+    public struct UnsafeArray<T> : IDisposable,
```

Comment:
@mbaker3 

1. Should we use this vs `UnsafeList` and just remember to be careful? 
2. If we keep this, should we clean it up or keep it the same as Unity's `NativeArray`?